app/controllers/templates_controller.rb

```diff
@@ -87,11 +87,18 @@ def preview
     else
       @template = resource_class.new(params[type_name_plural])
     end
-    base = @template.class.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
-    if @host.nil?
-      render :plain => _('No host could be found for rendering the template'), :status => :not_found
-      return
+
+    template_kind = TemplateKind.find_by(id: params[:kind]) if params[:kind]
```

Comment:
we could name the param with the full `template_kind_id` at least, but that's entirely optional.